shard_consumer.go

```diff
@@ -106,6 +131,23 @@ func (k *Kinsumer) captureShard(shardID string) (*checkpointer, error) {
 	}
 }
 
+func (k *Kinsumer) getLastSequenceNumber(shardID string) (string, error) {
```

Comment:
I think similar logic exists elsewhere in the app so would like to avoid this function.

Happily though, if we move the retry up a level of scope, we already keep track of the last sequence number that we passed to the checkpointer, in the [`lastSeqToCheckp`](https://github.com/snowplow-devops/kinsumer/blob/c77820e5182268b42154b3dbbab46569f2e9c527/shard_consumer.go#L290) variable.

So, we can remove this entire function, and use that - which will also mean less risk of duplicates, because we would start with the last one confirmed processed, rather than the one in the DDB table, which might be out of date.